File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.py
class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        vec = [num for row in grid for num in row]
        vec_length = len(vec)

        # Find the median using nth_element equivalent: sort and pick middle
        vec.sort()
        target = vec[vec_length // 2]

        result = 0
        for num in vec:
            if num % x != target % x:
                return -1
            result += abs(target - num) // x

        return result

File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.js
var minOperations = function (grid, x) {
    const m = grid.length;
    const n = grid[0].length;

    const vec = [];
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            vec.push(grid[i][j]);
        }
    }
    const vec_length = vec.length;

    vec.sort((a, b) => a - b);

    const target = vec[Math.floor(vec_length / 2)];
    let result = 0;
    for (const num of vec) {
        if (num % x !== target % x)
            return -1;
        result += Math.abs(target - num) / x;
    }
    return result;
};